### PR TITLE
Clang minifixes

### PR DESCRIPTION
**Pull request by Kakadu**
Some warnings detected by clang compiler are fixed.

### CODE REVIEW

File: src/include/kernel/spinlock.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ typedef struct {
 /* XXX use 'field : value' instead of '.field = value' syntax because g++ does not support
  * the second one, but supports the first one in the trivial order --Alexander */
 #define SPIN_INIT(state) \
-  { l : state, owner : -1u, __SPIN_CONTENTION_FIELD_INIT }
+  { .l = state, .owner = -1u, __SPIN_CONTENTION_FIELD_INIT }
```
Comment (anton-bondarev) [inline src/include/kernel/spinlock.h]: This is required for C++ as described above

File: src/kernel/task/resource/sig_table.c [hunk 2]
```diff
@@ -32,6 +32,15 @@ static void task_sig_handler_ignore(int sig) {
 #define Core task_sig_handler_terminate /* FIXME */
 #define Stop task_sig_handler_terminate /* FIXME */
 #define Cont task_sig_handler_ignore    /* FIXME */
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winitializer-overrides"
```
Comment (anton-bondarev) [inline src/kernel/task/resource/sig_table.c]: This is very difficult can be make it easy?
If I right understand this problem, there are some syscall numbers with identical id for example 
```
#define SIGABRT     6       /* (ANSI)     Abort.  */
#define SIGIOT      6       /* (4.2 BSD)  IOT trap.  */
```
I think right way to fix this warning rewrite initialization array

**Comment by Kakadu** [comment]
I think that I fixed issue about designator fields.

**Comment by anton-bondarev** [comment]
@Kakadu Yes, thanks, Looks good!

**Pull request merged**